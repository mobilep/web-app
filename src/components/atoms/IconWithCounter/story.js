import React from 'react';
import { storiesOf } from '@storybook/react';
import { PrimaryNavigationItem, IconWithCounter } from '..';
import { number, text } from '@storybook/addon-knobs/react';
import StoryRouter from 'storybook-react-router';
import { ScenariousIcon } from '../../../icons';

storiesOf('1. Atoms / IconWithCounter', module)
	.addDecorator(StoryRouter())
	.add('default',
		() => (<div style={{ width: '64px', height: '850px' }}>
			<PrimaryNavigationItem to={text('to', '/someUrl')}>
				<IconWithCounter icon={<ScenariousIcon />} counter={number('Unread count', 4)} />
				<span>{text('text', 'INBOX')}</span>
			</PrimaryNavigationItem>
		</div>));
