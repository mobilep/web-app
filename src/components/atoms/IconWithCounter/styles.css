.IconWithCounter {
	position: relative;
}

.IconWithCounter-counter {
	position: absolute;
	top: 0;
	left: 15px;
	width: 12px;
	height: 12px;
	border-radius: 50%;
	background-color: var(--white);
	color: var(--tomatoRed);
	font-size: 8px;
	line-height: 12px;
	transition: all .2s ease-in-out;
	text-align: center;
}

.IconWithCounter-counter-animation {
	animation: bounce .08s ease-in-out 2 alternate;
}

@media (max-width: 768px) {
	.IconWithCounter-counter {
		background-color: var(--grey3);
		color: var(--white);
		right: auto;
		left: 50%;
		transform: translateX(calc(-50% + 8px));
	}
}

@keyframes bounce {
	from {
		transform: scale(1);
	}

	to {
		transform: scale(1.3);
	}
}
