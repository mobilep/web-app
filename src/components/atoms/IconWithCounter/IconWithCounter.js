import React, { createRef, PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './styles.css';

class IconWithCounter extends PureComponent {

	static propTypes = {
		counter: PropTypes.number,
		icon: PropTypes.element.isRequired,
	};

	state = {
		isChanging: true,
	};

	counterRef = createRef();

	genClassName = (state) => classNames('IconWithCounter-counter', {
		'IconWithCounter-counter-animation': state.isChanging,
	});

	handleAnimationEnd = () => {
		this.setState({ isChanging: false });
	}

	componentDidUpdate (prevProps) {
		/* eslint-disable-next-line react/no-did-update-set-state */
		if (prevProps.counter !== this.props.counter) this.setState({ isChanging: true });
	}

	render () {
		const { counter, icon } = this.props;
		const unreadCount = counter > 9 ? '9+' : counter;

		return (
			<div className='IconWithCounter'>
				{unreadCount
					? <div
						className={this.genClassName(this.state)}
						onAnimationEnd={this.handleAnimationEnd}
					>
						{unreadCount}
					</div>
					: null
				}
				{icon}
			</div>
		);
	}
}

export default IconWithCounter;
