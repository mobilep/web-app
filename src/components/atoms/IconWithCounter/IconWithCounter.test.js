import React from 'react';
import { shallow } from 'enzyme';

import IconWithCounter from './IconWithCounter';

describe('Atom - IconWithCounter', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<IconWithCounter
				counter={1}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should not show more than nine (9) unread inboxes', () => {
		const wrapper = shallow(
			<IconWithCounter
				counter={10}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should not show the counter when there are no unread messages', () => {
		const wrapper = shallow(
			<IconWithCounter
				counter={0}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
