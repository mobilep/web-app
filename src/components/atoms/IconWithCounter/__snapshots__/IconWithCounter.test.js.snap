// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - IconWithCounter Should not show more than nine (9) unread inboxes 1`] = `
<div
  className="IconWithCounter"
>
  <div
    className="IconWithCounter-counter IconWithCounter-counter-animation"
    onAnimationEnd={[Function]}
  >
    9+
  </div>
</div>
`;

exports[`Atom - IconWithCounter Should not show the counter when there are no unread messages 1`] = `
<div
  className="IconWithCounter"
/>
`;

exports[`Atom - IconWithCounter Should render 1`] = `
<div
  className="IconWithCounter"
>
  <div
    className="IconWithCounter-counter IconWithCounter-counter-animation"
    onAnimationEnd={[Function]}
  >
    1
  </div>
</div>
`;
